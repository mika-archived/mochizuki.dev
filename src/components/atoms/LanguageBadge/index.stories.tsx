import React from "react";
import styled from "styled-components";
import { storiesOf } from "@storybook/react";

import LanguageBadge from ".";

const Badge = styled(LanguageBadge)`
  width: 24px;
  height: 24px;
`;

storiesOf("atoms/LanguageBadge", module).add("default", () => (
  <>
    <Badge color="#178600" />
  </>
));
