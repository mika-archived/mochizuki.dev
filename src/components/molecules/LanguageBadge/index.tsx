import React from "react";
import styled from "styled-components";

import ColorCircle from "../../atoms/ColorCircle";
import { Small } from "../../atoms/Typography";

type Props = {
  color: string;
  language: string;
};

const FlexBox = styled.div`
  display: flex;
  flex-flow: row nowrap;
`;

const LanguageColor = styled(ColorCircle)`
  width: 16px;
  height: 16px;
`;

const LanguageName = styled(Small)`
  margin-left: 4px;
  line-height: 1.3;
  color: #888;
`;

const LanguageBadge: React.FC<Props> = ({ color, language }) => {
  return (
    <FlexBox>
      <LanguageColor color={color} />
      <LanguageName>{language}</LanguageName>
    </FlexBox>
  );
};

export default LanguageBadge;
