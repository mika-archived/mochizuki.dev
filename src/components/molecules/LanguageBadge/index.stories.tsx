import React from "react";
import { storiesOf } from "@storybook/react";

import LanguageBadge from ".";

storiesOf("molecules/LanguageBadge", module).add("default", () => (
  <>
    <LanguageBadge color="#178600" language="C#" />
  </>
));
