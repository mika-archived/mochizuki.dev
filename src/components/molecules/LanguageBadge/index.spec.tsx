import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import LanguageBadge from ".";

afterEach(cleanup);

test("does not have changes from snapshot", () => {
  const { asFragment } = render(<LanguageBadge color="#178600" language="C#" />);

  expect(asFragment()).toMatchSnapshot();
});
