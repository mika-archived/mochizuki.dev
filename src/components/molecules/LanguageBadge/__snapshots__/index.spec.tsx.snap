// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background-color: #178600;
  border-radius: 9999px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.c2 {
  width: 16px;
  height: 16px;
}

.c3 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  margin-left: 4px;
  line-height: 1.3;
  color: #888;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      color="#178600"
    />
    <p
      class="sc-AxirZ sc-AxhCb c3"
    >
      C#
    </p>
  </div>
</DocumentFragment>
`;
